### PR TITLE
Oracle issues

### PR DESCRIPTION
**Pull request by juliushaertl**
- [x] not null integer with default 0
- [x] Issue when creating a new groupfolder

```
[3718718] [Wed Nov 18 20:54:47 2020] {"Exception":"Doctrine\\DBAL\\Driver\\OCI8\\OCI8Exception","Message":"ORA-02289: sequence does not exist","Code":0,"Trace":[{"file":"\/home\/jus\/repos\/nextcloud\/oci\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/OCI8\/OCI8Statement.php","line":407,"function":"fromErrorInfo","class":"Doctrine\\DBAL\\Driver\\OCI8\\OCI8Exception","type":"::","args":[{"code":2289,"message":"ORA-02289: sequence does not exist","offset":7,"sqltext":"SELECT \"group_folders_SEQ\".CURRVAL FROM DUAL"}]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/OCI8\/OCI8Connection.php","line":126,"function":"execute","class":"Doctrine\\DBAL\\Driver\\OCI8\\OCI8Statement","type":"->","args":[]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/OCI8\/OCI8Connection.php","line":168,"function":"query","class":"Doctrine\\DBAL\\Driver\\OCI8\\OCI8Connection","type":"->","args":["SELECT \"group_folders_SEQ\".CURRVAL FROM DUAL"]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Connection.php","line":1556,"function":"lastInsertId","class":"Doctrine\\DBAL\\Driver\\OCI8\\OCI8Connection","type":"->","args":["\"group_folders_SEQ\""]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/private\/DB\/Connection.php","line":270,"function":"lastInsertId","class":"Doctrine\\DBAL\\Connection","type":"->","args":["\"group_folders_SEQ\""]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/private\/DB\/AdapterOCI8.php","line":38,"function":"realLastInsertId","class":"OC\\DB\\Connection","type":"->","args":["\"group_folders_SEQ\""]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/private\/DB\/Connection.php","line":265,"function":"lastInsertId","class":"OC\\DB\\AdapterOCI8","type":"->","args":["\"group_folders_SEQ\""]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/apps\/groupfolders\/lib\/Folder\/FolderManager.php","line":379,"function":"lastInsertId","class":"OC\\DB\\Connection","type":"->","args":["group_folders"]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/apps\/groupfolders\/lib\/Controller\/FolderController.php","line":81,"function":"createFolder","class":"OCA\\GroupFolders\\Folder\\FolderManager","type":"->","args":["Test1"]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":218,"function":"addFolder","class":"OCA\\GroupFolders\\Controller\\FolderController","type":"->","args":["Test1"]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":127,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\GroupFolders\\Controller\\FolderController"},"addFolder"]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/private\/AppFramework\/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\GroupFolders\\Controller\\FolderController"},"addFolder"]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/private\/Route\/Router.php","line":309,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\GroupFolders\\Controller\\FolderController","addFolder",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"action":null,"_route":"groupfolders.Folder.addFolder"}]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/lib\/base.php","line":1008,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/groupfolders\/folders"]},{"file":"\/home\/jus\/repos\/nextcloud\/oci\/index.php","line":37,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/home\/jus\/repos\/nextcloud\/oci\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/OCI8\/OCI8Exception.php","Line":25,"CustomMessage":"--"}
```

### CODE REVIEW

**Comment by juliushaertl** [comment]
:ping_pong: for reviews :wink:

File: lib/Migration/Version401001Date20190715092137.php [hunk 1]
```diff
@@ -54,7 +54,6 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 				'length' => 64,
 			]);
 			$table->setPrimaryKey(['folder_id', 'mapping_type', 'mapping_id']);
-			$table->addUniqueIndex(['folder_id', 'mapping_type', 'mapping_id'], 'groups_folder_manage_unique');
```
Comment (nickvergessen) [inline lib/Migration/Version401001Date20190715092137.php:57]: :see_no_evil:

Comment (nickvergessen) [reply lib/Migration/Version401001Date20190715092137.php:57]: Drop in a migration?

File: lib/Migration/Version802000Date20201119112624.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
```
Comment (nickvergessen) [inline lib/Migration/Version802000Date20201119112624.php:4]: no copyright header?

**Review by nickvergessen — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**